src/v/coproc/reconciliation_backend.cc

```diff
@@ -362,8 +363,13 @@ reconciliation_backend::create_non_replicable_partition(
               ? errc::success
               : errc::partition_not_exists;
         }
+        // Non-replicated topics are not integrated with shadow indexing yet,
+        // so it's safe to not provide correct initial_revision_id.
         auto ntp_cfg = tt_md->second.get_configuration().cfg.make_ntp_config(
-          _data_directory, ntp.tp.partition, rev);
+          _data_directory,
+          ntp.tp.partition,
+          rev,
+          model::initial_revision_id(rev()));
```

Comment:
nit: I think it would be better to either pass correct initial revision id here (so that we can get rid of comment), or pass an obviously invalid revision id.

Reply:
fixed